test/framework/src/main/java/org/opensearch/test/OpenSearchIntegTestCase.java

```diff
@@ -2349,4 +2358,88 @@ protected ClusterState getClusterState() {
         return client(internalCluster().getClusterManagerName()).admin().cluster().prepareState().get().getState();
     }
 
+    /**
+     * Refreshes all indices in the cluster when forceRefresh set to true and
+     * checks if replica shards caught up with primary shard when Segment Replication is enabled.
+     */
+    protected void waitForReplication(boolean forceRefresh, String... indices) {
+        if (forceRefresh) {
+            refresh(indices);
+        }
+        if (isInternalCluster()) {
+            if (indices.length == 0) {
+                indices = getClusterState().routingTable().indicesRouting().keySet().toArray(String[]::new);
+            }
+            try {
+                for (String index : indices) {
+                    IndexRoutingTable indexRoutingTable = getClusterState().routingTable().index(index);
+                    if (indexRoutingTable != null) {
+                        assertBusy(() -> {
+                            for (IndexShardRoutingTable shardRoutingTable : indexRoutingTable) {
+                                final ShardRouting primaryRouting = shardRoutingTable.primaryShard();
+                                if (primaryRouting.state().toString().equals("STARTED")) {
+                                    final String indexName = primaryRouting.getIndexName();
+                                    if (isSegmentReplicationEnabledForIndex(indexName)) {
+                                        final List<ShardRouting> replicaRouting = shardRoutingTable.replicaShards();
+                                        final IndexShard primaryShard = getIndexShard(primaryRouting, indexName);
+                                        final Map<String, StoreFileMetadata> primarySegmentMetadata = primaryShard.getSegmentMetadataMap();
+                                        for (ShardRouting replica : replicaRouting) {
+                                            if (replica.state().toString().equals("STARTED")) {
+                                                IndexShard replicaShard = getIndexShard(replica, indexName);
+                                                final Store.RecoveryDiff recoveryDiff = Store.segmentReplicationDiff(
```

Comment:
This is doing more than waiting until its up to date.  It was verifying the cksums of all files across primary/replica.  I think for the purpose of "wait" you only need to ensure the SegmentInfos versions are the same between pri/rep.

Reply:
On that note, I think the verify method should probably stay as is?

Reply:
Got it, will update it in next commit